### ISSUE TITLE
Fix default pointer initialization of derived type components

### ISSUE THREAD

**Issue by mleair** [flang-compiler/f18-llvm-project#946]
This issue was first noticed while working on issue #921. Flang does not perform the default pointer initialization of the derived type in the following example:

```fortran
MODULE MUMPS_FRONT_DATA_MGT_M
      IMPLICIT NONE
      PRIVATE

      PUBLIC :: MUMPS_FDM_STRUC_TO_MOD

      TYPE FDM_STRUC_T
        INTEGER :: NB_FREE_IDX
        INTEGER, DIMENSION(:), POINTER :: STACK_FREE_IDX => null()
        INTEGER, DIMENSION(:), POINTER :: COUNT_ACCESS   => null()
      END TYPE FDM_STRUC_T

      TYPE (FDM_STRUC_T), TARGET, SAVE :: FDM_F

      CONTAINS

      SUBROUTINE MUMPS_FDM_STRUC_TO_MOD(id_FDM_ENCODING)

      CHARACTER, DIMENSION(:), POINTER :: id_FDM_ENCODING
      FDM_F = transfer(id_FDM_ENCODING,FDM_F)
      DEALLOCATE(id_FDM_ENCODING)
      NULLIFY(id_FDM_ENCODING)

      RETURN
      END SUBROUTINE MUMPS_FDM_STRUC_TO_MOD

      END MODULE MUMPS_FRONT_DATA_MGT_M
```

**Comment by jeanPerier** [comment]
This is hitting the TODO: not yet implemented default initialization of global variables. Implementation is in the pipeline.

**Comment by oroppas** [comment]
#945

**Comment by jeanPerier** [comment]
Appears fixed to me by #963. I can see that the LLVM IR gloabal definition for `FDM_F` is initializing the pointer descriptors component to NULL :

```
@_QMmumps_front_data_mgt_mEfdm_f = linkonce global %_QMmumps_front_data_mgt_mTfdm_struc_t {
  i32 undef,
  { i32*, i64, i32, i8, i8, i8, i8, [1 x [3 x i64]] } { i32* null, i64 4, i32 20180515, i8 1, i8 9, i8 1, i8 0, [1 x [3 x i64]] [[3 x i64] [i64 1, i64 0, i64 4]] },
  { i32*, i64, i32, i8, i8, i8, i8, [1 x [3 x i64]] } { i32* null, i64 4, i32 20180515, i8 1, i8 9, i8 1, i8 0, [1 x [3 x i64]] [[3 x i64] [i64 1, i64 0, i64 4]] } }
```
